Hymns/Display/DisplayHymnViewModel.swift

```diff
@@ -32,36 +40,64 @@ class DisplayHymnViewModel: ObservableObject {
         self.historyStore = historyStore
         hymnLyricsViewModel = HymnLyricsViewModel(hymnToDisplay: identifier)
     }
-
+    
     deinit {
         favoritesObserver = nil
     }
-
+    
     func fetchHymn() {
         repository
             .getHymn(identifier)
-            .map({ [weak self] hymn -> String? in
-                guard let self = self else { return nil }
-                if self.identifier.hymnType == .classic {
-                    return "Hymn \(self.identifier.hymnNumber)"
-                }
+            .map({ [weak self] hymn -> (Title?, Lyrics?, MetaDatum?) in
+                guard let self = self else { return (nil, nil, nil) }
                 guard let hymn = hymn else {
-                    return nil
+                    return (nil, nil, nil)
                 }
-                return hymn.title.replacingOccurrences(of: "Hymn: ", with: "")
+                if self.identifier.hymnType == .classic {
+                    return ("Hymn \(self.identifier.hymnNumber)", hymn.lyrics, hymn.pdfSheet)
+                }
+                return (hymn.title.replacingOccurrences(of: "Hymn: ", with: ""), hymn.lyrics, hymn.pdfSheet)
```

Comment:
I made one tiny tweak by removing this map, but other than that, it looks perfect. Great job!